src/main/java/org/ladysnake/blabber/impl/common/model/DialogueIllustrationSelectorEntity.java

```diff
@@ -0,0 +1,125 @@
+
+import com.mojang.brigadier.StringReader;
+import com.mojang.brigadier.exceptions.CommandSyntaxException;
+import com.mojang.datafixers.util.Either;
+import com.mojang.datafixers.util.Pair;
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.font.TextRenderer;
+import net.minecraft.client.gui.DrawContext;
+import net.minecraft.client.gui.screen.ingame.InventoryScreen;
+import net.minecraft.command.EntitySelector;
+import net.minecraft.command.EntitySelectorReader;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.network.PacketByteBuf;
+import net.minecraft.server.command.ServerCommandSource;
+import net.minecraft.util.dynamic.Codecs;
+import org.jetbrains.annotations.Nullable;
+import org.ladysnake.blabber.api.DialogueIllustration;
+import org.ladysnake.blabber.api.DialogueIllustrationType;
+
+import java.util.Optional;
+import java.util.UUID;
+
+public record DialogueIllustrationSelectorEntity(Either<String, LivingEntity> selector, int x, int y, int x2, int y2,
+                                                 int size, float yOff, Optional<Integer> stareAtX,
+                                                 Optional<Integer> stareAtY) implements DialogueIllustration {
+    private static final Codec<DialogueIllustrationSelectorEntity> CODEC = RecordCodecBuilder.create(instance -> instance.group(
+            Codec.STRING.fieldOf("entity")
+                    .xmap(Either::<String, LivingEntity>left, e -> e.map(s -> s, Entity::getUuidAsString))
+                    .forGetter(DialogueIllustrationSelectorEntity::selector),
+            Codec.INT.fieldOf("x").forGetter(DialogueIllustrationSelectorEntity::x),
+            Codec.INT.fieldOf("y").forGetter(DialogueIllustrationSelectorEntity::y),
+            Codec.INT.fieldOf("x2").forGetter(DialogueIllustrationSelectorEntity::x2),
+            Codec.INT.fieldOf("y2").forGetter(DialogueIllustrationSelectorEntity::y2),
+            Codec.INT.fieldOf("size").forGetter(DialogueIllustrationSelectorEntity::size),
+            Codecs.createStrictOptionalFieldCodec(Codec.FLOAT, "y_offset", 0.0f).forGetter(DialogueIllustrationSelectorEntity::yOff),
+            Codecs.createStrictOptionalFieldCodec(Codec.INT, "stare_at_x").forGetter(DialogueIllustrationSelectorEntity::stareAtX),
+            Codecs.createStrictOptionalFieldCodec(Codec.INT, "stare_at_y").forGetter(DialogueIllustrationSelectorEntity::stareAtY)
+    ).apply(instance, DialogueIllustrationSelectorEntity::new));
+
+    @Override
+    public void render(DrawContext context, TextRenderer textRenderer, int x, int y, int mouseX, int mouseY, float tickDelta) {
+        selector.ifRight(e -> {
+            int fakedMouseX = stareAtX.map(s -> s + x + this.x).orElse(mouseX);
+            int fakedMouseY = stareAtY.map(s -> s + y + this.y).orElse(mouseY);
```

Comment:
Not sure the "staring at" coordinates should take the illustrations' own positions into account - as in, if you want two entities that look at each other, it would be easier if you could make them both "stare at" some point in the middle.

Reply:
This does make it easier to make something stare relative to the box itself, though.

Reply:
Possibly yes... Although I'd say, if we're going to make those coordinates relative to the illustration, I think (0, 0) should be the center